Improvements to handling of nested classes: correctly populate type

wmdietlGC opened the pull request:
arguments of enclosing classes, output enclosing classes in error
messages, and enable and add related tests.

smillst commented on framework/tests/all-systems/Issue263.java:
```diff
@@ -1,8 +1,6 @@
 
-// Suppression required because of Issue 724
-// https://github.com/typetools/checker-framework/issues/724
 @SuppressWarnings({"regex", "tainting"})
```
Should the warning suppress be removed now?

wmdietlGC replied:
Yes, I don't know why I only removed the comments.

wmdietlGC replied:
Removing the SuppressWarnings works.

wmdietlGC commented:
Which comment do you mean with 'Does this pull request fix 724? If so, this comment in AnnotatedTypes#outerAsSuper needs to be addressed:'
The link seems to point to all changes.

wmdietlGC commented:
I found this one: https://github.com/typetools/checker-framework/blob/master/framework/src/org/checkerframework/framework/util/AnnotatedTypes.java#L185
Let me try whether that can also be removed.

wmdietlGC commented:
Removing https://github.com/typetools/checker-framework/blob/master/framework/src/org/checkerframework/framework/util/AnnotatedTypes.java#L185 doesn't work unfortunately.
So I guess some of the cases of Issue #724 are fixed, but not all.
This PR still fixes a few actual crashes, so I would argue for merging it.

smillst commented:
These changes look correct to me and should be merged once the test pass.  Is #134 fixed by these changes?

wmdietlGC commented:
I added a test for #134 and for #437, uncommented some other tests, and added my own test that made me interested in this issue.
Some parts of #724 also work now, but https://github.com/typetools/checker-framework/blob/master/framework/src/org/checkerframework/framework/util/AnnotatedTypes.java#L185 can't be removed yet. I'll add a comment to that effect to #724, but leave it open.
I'm still debugging a few other crashes that are also related to inner classes, so more will come.
